tests/xtro-sharpie/SimdCheck.cs

```diff
@@ -0,0 +1,382 @@
+			// and if there are any, then assume the method binding has a [MarshalDirective].
+			var body = method.Body;
+			var anyCalls = false;
+			foreach (var i in body.Instructions) {
+				switch (i.OpCode.Code) {
+				case Mono.Cecil.Cil.Code.Call:
+				case Mono.Cecil.Cil.Code.Calli:
+				case Mono.Cecil.Cil.Code.Callvirt:
+					var mr = i.Operand as MethodReference;
+					if (mr != null) {
+						if (mr.Name.StartsWith ("xamarin_simd__", StringComparison.Ordinal))
+							return;
+						if (mr.Name.StartsWith ("xamarin_vector_float3__", StringComparison.Ordinal))
+							return;
+					}
+					anyCalls = true;
+					break;
+				default:
+					break;
+				}
+			}
+
+			// If the method doesn't call anywhere, it can't be broken.
+			// For instance if the method just throws an exception.
+			if (!anyCalls)
+				return;
+
+			Console.WriteLine ($"!simd-missing-marshaldirective! {method}: simd type: {simd_type}");
+		}
+
+		public override void VisitObjCMethodDecl (ObjCMethodDecl decl, VisitKind visitKind)
+		{
+			if (visitKind != VisitKind.Enter)
+				return;
+
+			// don't process methods (or types) that are unavailable for the current platform
+			if (!decl.IsAvailable () || !(decl.DeclContext as Decl).IsAvailable ())
+				return;
+
+			var simd_type = string.Empty;
+			var requires_marshal_directive = false;
+			var native_simd = ContainsSimdTypes (decl, ref simd_type, ref requires_marshal_directive);
+
+			ManagedSimdInfo info;
+			managed_methods.TryGetValue (decl.GetName (), out info);
+			var method = info?.Method;
+
+			if (!native_simd) {
+				if (method != null) {
+					// The managed method uses types that were incorrectly used in place of the correct Simd types,
+					// but the native method doesn't use the native Simd types. This means the binding is correct.
+				} else {
+					// Neither the managed nor the native method have anything to do with Simd types.
+				}
+				return;
+			}
+
+			if (method == null) {
+				// Could not map the native method to a managed method.
+				// This needs investigation, to see why the native method couldn't be mapped.
+
+				// Check if this is new API, in which case it probably couldn't be mapped because we haven't bound it.
+				var is_new = false;
+				var attrs = decl.Attrs.ToList ();
+				var parentClass = decl.DeclContext as Decl;
+				if (parentClass != null)
+					attrs.AddRange (parentClass.Attrs);
+
+				foreach (var attr in attrs) {
+					var av_attr = attr as AvailabilityAttr;
+					if (av_attr == null)
+						continue;
+					if (av_attr.Platform.Name != "ios")
+						continue;
+					if (av_attr.Introduced.Major >= 11) {
+						is_new = true;
+						break;
+					}
+				}
+				if (is_new && !very_strict)
+					return;
+				if (!strict)
+					return;
+				Console.WriteLine ($"!simd-can't-map-native! {decl}: could not find a managed method (selector: {decl.Selector} name: {decl.GetName ()}. Found the simd type '{simd_type}' in the native signature.");
```

Comment:
start with either `!unknown-` or `!wrong-` so it gets sorted along other important/critical logs